extensions/amp-bind/0.1/bind-impl.js

```diff
@@ -912,6 +938,50 @@ export class Bind {
   }
 
   /**
+   * Deep merge object b into object a. Both a and b can only contain
+   * primitives, arrays, and objects created by Object.create(null).
+   * Arrays are replaced, not merged. Other objects are recursively merged.
+   * @param {!Object} a the destination object
+   * @param {!Object} b
+   * @param {number=} opt_depth The depth of the objects being merged
+   * @return {!Object}
+   */
+  deepMerge_(a, b, opt_depth) {
+    const depth = opt_depth || 0;
+    if (depth > this.maxMergeDepth_) {
+      user().error(TAG, 'merge depth exeeds limit');
+      Object.assign(a, b);
+      return a;
+    }
+    Object.keys(b).forEach(key => {
+      const newValue = b[key];
+      // Perform a deep merge IFF
+      // 1: Both a and b have the same property
+      if (Object.hasOwnProperty.call(a, key)) {
```

Comment:
We have a [helper](https://github.com/ampproject/amphtml/blob/20b2edb5813889a25ad03238e46a5373efa3aea6/src/utils/object.js#L36-L46) for this.